app/models/user.rb

```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
+
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  
+  has_one :profile
+
+  validates :email, presence: true
```

Comment:
A cool trick that Rails has is a format validation for email. You can either use a regex for it or you can use `, format: { with: URI::MailTo::EMAIL_REGEXP }`, which will automatically raise an error if you input something that is not a valid email

Reply:
final version would look like `validates :email, presence: true, format: { with: URI::MailTo::EMAIL_REGEXP }`